analyzers/src/SonarAnalyzer.CSharp/Rules/SetLocaleForDataTypes.cs

```diff
@@ -118,15 +135,105 @@ public sealed class SetLocaleForDataTypes : SonarDiagnosticAnalyzer
             {
                 AssignmentExpressionSyntax assignment => assignment.Left,
                 EqualsValueClauseSyntax equalsClause when equalsClause.Parent.Parent is VariableDeclarationSyntax variableDeclaration => variableDeclaration.Variables.Last(),
+                ArgumentSyntax argument => HandleArgumentSyntax(argument),
                 _ => null
             };
 
-        private static IPropertySymbol GetPropertySymbol(AssignmentExpressionSyntax assignment, SemanticModel model) =>
-            model.GetSymbolInfo(assignment.Left).Symbol as IPropertySymbol;
+        private static SyntaxNode HandleArgumentSyntax(ArgumentSyntax argument)
+        {
+            if (TupleExpressionSyntaxWrapper.IsInstance(argument.Parent)
+                && ((TupleExpressionSyntaxWrapper)argument.Parent) is var rightTuple)
+            {
+                var currentNode = rightTuple.SyntaxNode.Parent;
+                AssignmentExpressionSyntax assignment = null;
+                while (currentNode != null && assignment == null)
+                {
+                    if (currentNode is AssignmentExpressionSyntax syntax)
+                    {
+                        assignment = syntax;
+                    }
+                    else
+                    {
+                        currentNode = currentNode.Parent;
+                    }
+                }
+                return FindAssignedSyntaxNodeInAssignment(argument, assignment);
+            }
+            return null;
+        }
+
+        private static SyntaxNode FindAssignedSyntaxNodeInAssignment(ArgumentSyntax argument, AssignmentExpressionSyntax assignment)
```

Comment:
Consider using [IDeconstructionAssignmentOperation](https://docs.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.operations.ideconstructionassignmentoperation?view=roslyn-dotnet-3.2.1) for this. I didn't look into the details, but it has `Target` and `Value`. I also don't know how good the IOperation part of the shim layer is.

Reply:
This method  does not exist anymore.